lib/server/roster.ts

```diff
@@ -0,0 +1,221 @@
+import "server-only";
+
+import { auth } from "@clerk/nextjs/server";
+import { eq } from "drizzle-orm";
+
+import {
+  roster,
+  rosterArtifacts,
+  rosterHeroes,
+  rosterLevels,
+} from "@/drizzle/schema";
+import { drizzleClient } from "@/lib/server/drizzle";
+import { AscensionLevel } from "../characters";
+
+const drizzle = drizzleClient;
+
+export async function createOrUpdateRoster(
+  userId: string,
+): Promise<string | false> {
```

Comment:
i think instead of returning `false` this should `throw` an exception

Reply:
So sure I can throw an exception... but ultimately the route doesn't care... we aren't going to show the exception error to the user, so do we care? Or is this just for backend server logs and you can see when something throws?

Reply:
its for the backend, and also it helps with the type system downstream. you don't need to handle both `string` and `false` cases in the code that uses this method

Reply:
Done!